libs/service/src/config.cpp

```diff
@@ -0,0 +1,210 @@
+
+#include <atomic>
+#include <filesystem>
+#include <fstream>
+#include <functional>
+#include <sstream>
+
+#include "config.h"
+#include "mapget/log.h"
+
+namespace mapget
+{
+
+DataSourceConfigService& DataSourceConfigService::get()
+{
+    static DataSourceConfigService instance;
+    return instance;
+}
+
+DataSourceConfigService::Subscription::~Subscription()
+{
+    DataSourceConfigService::get().unsubscribe(id_);
+}
+
+std::unique_ptr<DataSourceConfigService::Subscription> DataSourceConfigService::subscribe(
```

Comment:
Since the subscription mechanism identification is based on `id_`, it can be returned by value.

Reply:
You mean unique_ptr? Yes, exactly.

Reply:
You should remove the copy ctors of `DataSourceConfigService::Subscription` even if only used via unique_ptr.